extensions/google-drive/src/index.tsx

```diff
@@ -104,87 +104,183 @@ export default function Command() {
+
+      setFiles((prevFiles) => ({
+        ...prevFiles,
+        filtered: prevFiles.filtered.map((f) => (f.path === file.path ? updatedFile : f)),
+        favorites: isEmpty(debouncedSearchText) ? queryFavoriteFiles(db) : prevFiles.favorites,
+      }));
+    },
+    [db, setFiles, debouncedSearchText]
+  );
+
+  const handleSelectionChange = useDebouncedCallback((file: FileInfo | null) => {
+    setFiles((prevFiles) => ({ ...prevFiles, selected: file }));
+  }, 100);
+
   return (
-    <List
-      isShowingDetail={filesFiltered.length > 0}
-      enableFiltering={false}
-      onSearchTextChange={setSearchText}
-      searchBarPlaceholder={`Fuzzy search in ${displayPath(drivePath)}`}
-      isLoading={isFetching}
-      onSelectionChange={(id) => setSelectedFile(findFile(id))}
-    >
-      {filesFiltered.length > 0 ? (
-        <List.Section
-          title="Files"
-          subtitle={filesIndexGeneratedAt ? `Indexed At: ${filesIndexGeneratedAt.toLocaleString()}` : ""}
-        >
-          {filesFiltered.map((file) => (
-            <List.Item
-              id={file.displayPath}
-              key={file.displayPath}
-              icon={{ fileIcon: file.path }}
-              title={file.name}
-              detail={<List.Item.Detail markdown={fileDetailsMarkup} />}
-              actions={
-                <ActionPanel>
-                  <Action.Open title="Open File" icon={Icon.Document} target={file.path} />
-                  <Action.ShowInFinder path={file.path} />
-                  <Action.OpenWith path={file.path} />
-                  <Action.CopyToClipboard
-                    title="Copy File Path"
-                    content={escapePath(file.displayPath)}
-                    shortcut={{ modifiers: ["cmd", "shift"], key: "f" }}
-                  />
-                  <Action.CopyToClipboard
-                    title="Copy Folder Path"
-                    content={escapePath(dirname(file.displayPath))}
-                    shortcut={{ modifiers: ["cmd", "shift"], key: "d" }}
-                  />
-                  <ReindexFilesCacheAction reindexFiles={reindexFiles} />
-                  <ClearFilePreviewsCacheAction />
-                </ActionPanel>
+    <CommandContext.Provider value={{ handleToggleFavorite, reindexFiles, fileDetailsMarkup }}>
+      <List
+        isShowingDetail={files.filtered.length > 0}
+        enableFiltering={false}
+        onSearchTextChange={setSearchText}
+        searchBarPlaceholder={`Fuzzy search in ${displayPath(drivePath)}`}
+        isLoading={isFetching}
+        onSelectionChange={(id) => handleSelectionChange(findFile(id?.replace(/__.+__/, "")))}
+      >
+        {files.filtered.length > 0 ? (
+          <>
+            {isEmpty(debouncedSearchText) && files.favorites.length > 0 ? (
+              <List.Section title="Favorites">
+                {files.favorites.map((file) => (
+                  <ListItem idPrefix="__fav__" key={file.displayPath} file={file} />
```

Comment:
`idPrefix` is just to set a unique `id` for a favorite item otherwise it blows up due to duplicate `id` error when the same item is shown in both `Favorites` and `Recent` sections at the same time.